tidy3d/plugins/dispersion/fit.py

```diff
@@ -164,6 +167,44 @@ def __init__(self, wvl_um: Numpy, n_data: Numpy, k_data: Numpy = None):
         self.freqs = C_0 / wvl_um
         self.frequency_range = (np.min(self.freqs), np.max(self.freqs))
 
+    def set_wvl_range(self, wvl_min: float, wvl_max: float):
+        """Truncate the wavelength-nk data to wavelength range [wvl_min,wvl_max] for
+            fitting. Data outside this range will be ignored.
+
+        Parameters
+        ----------
+        wvl_min : float
+            The beginning of wavelength range. Unit: micron
+        wvl_max : float
+            The end of wavelength range. Unit: micron
+        """
+
+        if wvl_min < np.min(self.wvl_um):
+            log.warning(
+                "wvl_min is smaller than the minimal wavelength provided "
+                "in the data. Please increase wvl_min for accuracy."
+            )
+        if wvl_max > np.max(self.wvl_um):
+            log.warning(
+                "wvl_max is larger than the maximal wavelength provided "
+                "in the data. Please decrease wvl_max for accuracy."
+            )
+
+        ind_min = self.wvl_um >= wvl_min
+        ind_max = self.wvl_um <= wvl_max
+        ind_final = np.logical_and(ind_min, ind_max)
+
+        # update
+        self.wvl_um = self.wvl_um[ind_final]
+        self.n_data = self.n_data[ind_final]
+        self.k_data = self.k_data[ind_final]
+
+        if np.all(ind_final) is False:
+            log.error("No data within [wvl_min,wvl_max]")
+            return
+
+        self.__init__(self.wvl_um, self.n_data, self.k_data)
```

Comment:
is this supposed to return? I'm confused

Reply:
I suppose it just updates wvl_um, n_data, etc., but doesn't return anything.

Reply:
aren't the values updated above?  is the call to init needed?

Reply:
there are several others to be updated, such as frequency_range, eps_data, etc.

Reply:
removed